docs/layering.md

```diff
@@ -0,0 +1,369 @@
+This tutorial is based on a previously discussed [request](https://github.com/coreos/fedora-coreos-tracker/issues/1029).
+
+### **Note: This is an experimental feature that is subject to change.**
+
+To follow this guide you should have the prerequisites described in the [CoreOS documentation](https://docs.fedoraproject.org/en-US/fedora-coreos/tutorial-setup/).
+
+Let's first deploy a FCOS environment and do an initial config with Ignition. We are aiming for a simple deployment which is [described here](https://docs.fedoraproject.org/en-US/fedora-coreos/tutorial-autologin/).
+
+### Download Fedora CoreOS next stream
+
+We depend on some newer rpm-ostree and skopeo functionality.
+Let's use the next stream.
+
+```bash
+# This is the 'next' image we will be using.
+# Anything newer than this should work too.
+$ RELEASE="35.20211215.1.0"
+
+# Download the image.
+$ curl -O https://builds.coreos.fedoraproject.org/prod/streams/next/builds/$RELEASE/x86_64/fedora-coreos-$RELEASE-qemu.x86_64.qcow2.xz
+
+# Uncompress the qemu image.
+$ unxz fedora-coreos-$RELEASE-qemu.x86_64.qcow2.xz
+
+# Rename the file for easier handling.
+$ mv fedora-coreos-$RELEASE-qemu.x86_64.qcow2.xz fedora-coreos.qcow2
+```
+
+### First Ignition config via Butane
+
+Let's create a very simple Butane config that will perform the following actions:
+
+- Add a systemd drop-in to override the default `serial-getty@ttyS0.service`.
+    - The override will make the service automatically log the `core` user in to the serial console of the booted machine.
+- Set the system hostname by dropping a file at `/etc/hostname`,
+- Add a `/etc/cachefilesd.conf` file with our desired values.
+
+For this, we create a Butane configuration file `autologin.bu` with the following content:
+
+```yaml=
+variant: fcos
+version: 1.4.0
+systemd:
+  units:
+    - name: serial-getty@ttyS0.service
+      dropins:
+      - name: autologin-core.conf
+        contents: |
+          [Service]
+          # Override ExecStart in main unit
+          ExecStart=
+          # Add new Execstart with `-` prefix to ignore failure`
+          ExecStart=-/usr/sbin/agetty --autologin core --noclear %I $TERM
+storage:
+  files:
+    - path: /etc/hostname
+      mode: 0644
+      contents:
+        inline: |
+          tutorial
+    - path: /etc/cachefilesd.conf
+      mode: 0644
+      contents:
+        inline: |
+          dir /var/cache/fscache
+          tag mycache
+          brun  8%
+          bcull 6%
+          bstop 2%
+          frun  9%
+          fcull 5%
+          fstop 4%
+          secctx system_u:system_r:cachefiles_kernel_t:s0
+```
+
+This configuration can then be converted into an Ignition config with Butane:
+
+```bash
+$ butane --pretty --strict autologin.bu --output autologin.ign
+```
+
+The resulting Ignition configuration produced by Butane as `autologin.ign` has the following content:
```

Comment:
Do we need this bit?  Seems like noise versus just saying e.g. "Run `cat autologin.ign` to see the rendered config." or so.

Reply:
removed.